org.eclipse.xtext.ide.tests/src/org/eclipse/xtext/ide/tests/server/WorkspaceFoldersTest.xtend

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+ * Copyright (c) 2020 TypeFox GmbH (http://www.typefox.io) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.ide.tests.server
+
+import com.google.inject.AbstractModule
+import com.google.inject.Inject
+import com.google.inject.Module
+import com.google.inject.Scopes
+import java.io.File
+import java.io.FileWriter
+import java.util.concurrent.CompletableFuture
+import org.eclipse.lsp4j.DidChangeWorkspaceFoldersParams
+import org.eclipse.lsp4j.WorkspaceFolder
+import org.eclipse.lsp4j.WorkspaceFoldersChangeEvent
+import org.eclipse.xtext.ide.server.UriExtensions
+import org.eclipse.xtext.ide.server.WorkspaceManager
+import org.eclipse.xtext.util.Files
+import org.eclipse.xtext.util.Modules2
+import org.junit.Test
+
+import static org.junit.Assert.assertEquals
+import org.junit.rules.TemporaryFolder
+
+/**
+ * @author Jan Koehnlein - Initial contribution and API
+ */
+class WorkspaceFoldersTest extends AbstractTestLangLanguageServerTest {
+
+	@Inject extension UriExtensions
+
+	@Inject WorkspaceManager workspaceManager
+
+	@Inject extension TemporaryFolder 
```

Comment:
Must be a public field and annotated with `@Rule` to take effect.
The Javadoc of TemporaryFolder has an example.

Reply:
Fixed. Thanks